modules/appmodel/tests/test_app_transform_generator.c

```diff
@@ -0,0 +1,470 @@
+                                                              ),
+                                              create_transform("transform-2",
+                                                  create_step("step-2-1", "$MESSAGE = \"2/first step here\";"),
+                                                  create_step("step-2-2", "$MESSAGE += \"2/second step here\";"),
+                                                  NULL
+                                                              ),
+                                              NULL
+                                             );
+  _register_transformation(transform1->str);
+
+  _app_transform_generate("my-topic");
+
+  _assert_config_contains("case 'my-transformation':\n");
+  _assert_config_contains("$MESSAGE = \"first step here\";\n");
+  _assert_config_contains("$MESSAGE += \"second step here\";\n");
+  _assert_config_contains("$MESSAGE = \"2/first step here\";\n");
+  _assert_config_contains("$MESSAGE += \"2/second step here\";\n");
+  _assert_config_is_valid("my_topic", "meta", "");
+}
+
+Test(app_transform_generator, add_multiple_tranformations_with_particular_topic)
+{
+  GString *transform1 = create_transformation("my-transformation", "my-topic",
+                                              create_transform("transform-1",
+                                                  create_step("step-1-1", "$MESSAGE = \"1/1/first step here\";"),
+                                                  create_step("step-1-2", "$MESSAGE += \"1/1/second step here\";"),
+                                                  NULL
+                                                              ),
+                                              create_transform("transform-2",
+                                                  create_step("step-2-1", "$MESSAGE = \"1/2/first step here\";"),
+                                                  create_step("step-2-2", "$MESSAGE += \"1/2/second step here\";"),
+                                                  NULL
+                                                              ),
+                                              NULL
+                                             );
+  _register_transformation(transform1->str);
+
+  GString *transform2 = create_transformation("my-transformation2", "my-topic",
+                                              create_transform("transform-2/1",
+                                                  create_step("step-2-1-1", "$MESSAGE = \"2/1/first step here\";"),
+                                                  create_step("step-2-1-2", "$MESSAGE += \"2/1/second step here\";"),
+                                                  NULL
+                                                              ),
+                                              create_transform("transform-2/2",
+                                                  create_step("step-2-2-1", "$MESSAGE = \"2/2/first step here\";"),
```

Comment:
I think it's not worth hiding the the axosyslog config syntax behind these functions, tests would look simpler if we saw the full config snippet.